### PR TITLE
Enslige forsorgere

### PR DESCRIPTION
**Pull request by dragutin-nav**
## Describe your changes

Fetch data about enslige forsorgere from team familie

## Trello ticket number and link

https://trello.com/c/UixBT04u

## Type of change

- [X] New feature (non-breaking change which adds functionality)

## Checklist before requesting a review
- [X] I have performed a self-review of my code
- [X] If it is a core feature, I have added thorough tests.
- [❓ maybe] We need to implement grafana analytics

### CODE REVIEW

File: src/main/java/no/nav/pto/veilarbportefolje/ensligforsorger/EnsligeForsorgereRepository.java [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+
+    public void lagreEnsligeForsorgereStonad(VedtakOvergangsstønadArbeidsoppfølging vedtakOvergangsstønadArbeidsoppfølging) {
+        if (vedtakOvergangsstønadArbeidsoppfølging.getStønadstype().toString().equals("OVERGANGSSTØNAD")) {
+            lagreOvergangsstonad(vedtakOvergangsstønadArbeidsoppfølging);
+        } else {
+            log.info("Vi støtter kun overgangstønad for enslige forsorgere. Fått: " + vedtakOvergangsstønadArbeidsoppfølging.getStønadstype());
+        }
+    }
+
+    public void lagreOvergangsstonad(VedtakOvergangsstønadArbeidsoppfølging vedtakOvergangsstønadArbeidsoppfølging) {
```
Comment (slovrid) [inline src/main/java/no/nav/pto/veilarbportefolje/ensligforsorger/EnsligeForsorgereRepository.java:73]:
```suggestion
    private void lagreOvergangsstonad(VedtakOvergangsstønadArbeidsoppfølging vedtakOvergangsstønadArbeidsoppfølging) {
```

**Comment by slovrid** [comment]
General comment about the `ef_` abbreviation: I prefer to write out such names in full. I fear that it might get confusing later, even if we use the full name elsewhere. I don't think we should be "afraid" of having too long names, since in general the drawbacks of confusing abbreviations outweigh the potential benefits.

File: src/main/java/no/nav/pto/veilarbportefolje/ensligforsorger/EnsligeForsorgereRepository.java [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+                preparedStatement.setString(1, periodeTypeConst);
+                return preparedStatement;
+            }, generatedKeyHolder);
+            return generatedKeyHolder.getKey().intValue();
+        } catch (Exception e) {
+            throw new RuntimeException("Kan ikke lagre vedtak periodetype " + e);
+        }
+    }
+
+    private Integer hentVettakAktivitetstype(String aktivitetsType) {
```
Comment (slovrid) [inline src/main/java/no/nav/pto/veilarbportefolje/ensligforsorger/EnsligeForsorgereRepository.java:207]:
```suggestion
    private Integer hentVedtakAktivitetstype(String aktivitetsType) {
```

File: src/main/java/no/nav/pto/veilarbportefolje/ensligforsorger/EnsligeForsorgereRepository.java [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+        return tryCacheFirst(vettakAktivitetstypeCache, aktivitetsType, () -> this.hentVettakAktivitetstypeFraDB(aktivitetsType));
+    }
+
+    private Integer hentVettakAktivitetstypeFraDB(String aktivitetsType) {
```
Comment (slovrid) [inline src/main/java/no/nav/pto/veilarbportefolje/ensligforsorger/EnsligeForsorgereRepository.java:211]:
```suggestion
    private Integer hentVedtakAktivitetstypeFraDB(String aktivitetsType) {
```

File: src/main/java/no/nav/pto/veilarbportefolje/ensligforsorger/EnsligeForsorgereRepository.java [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+            .maximumSize(10)
+            .build();
+
+    private final Cache<String, Integer> vettakAktivitetstypeCache = Caffeine.newBuilder()
+            .expireAfterWrite(120, TimeUnit.MINUTES)
+            .maximumSize(10)
+            .build();
+
+    public void lagreEnsligeForsorgereStonad(VedtakOvergangsstønadArbeidsoppfølging vedtakOvergangsstønadArbeidsoppfølging) {
+        if (vedtakOvergangsstønadArbeidsoppfølging.getStønadstype().toString().equals("OVERGANGSSTØNAD")) {
```
Comment (slovrid) [inline src/main/java/no/nav/pto/veilarbportefolje/ensligforsorger/EnsligeForsorgereRepository.java:66]: `"OVERGANGSSTØNAD"` = candidate for an enum?

File: src/main/java/no/nav/pto/veilarbportefolje/opensearch/OpensearchQueryBuilder.java [hunk 5]
```diff
@@ -209,6 +206,12 @@ static void leggTilManuelleFilter(BoolQueryBuilder queryBuilder, Filtervalg filt
             );
             queryBuilder.must(avvik14aVedtakSubQuery);
         }
+
+        if (filtervalg.harEnsligeForsorgereFilter()) {
+            if (filtervalg.getEnsligeForsorgere().contains(EnsligeForsorgere.OVERGANGSSTØNAD)) {
```
Comment (slovrid) [inline src/main/java/no/nav/pto/veilarbportefolje/opensearch/OpensearchQueryBuilder.java:211]: Unnecessary nested `if` can be combined into single predicate.

File: src/main/resources/db/postgres/V1_62__enslige_forsorgere.sql [hunk 6]
```diff
@@ -0,0 +1,100 @@
+CREATE TABLE enslige_forsorgere_aktivitet_type
+(
+    id             SERIAL PRIMARY KEY,
```
Comment (slovrid) [inline src/main/resources/db/postgres/V1_62__enslige_forsorgere.sql:3]: Consider using generated id's, such as UUID?

**Review by slovrid — APPROVED** [review]
I'm approving this after having gotten a thorough run-through of the changes from Drago, where we discussed some tradeoffs that have been made. I gave some feedback during the review, in addition to a few comments I've left here. @dragutin-nav feel free to assess the feedback and whether you want to incorporate it into this PR.

File: src/main/java/no/nav/pto/veilarbportefolje/ensligforsorger/EnsligeForsorgereRepository.java (hunk 4 above)
Comment (dragutin-nav) [reply src/main/java/no/nav/pto/veilarbportefolje/ensligforsorger/EnsligeForsorgereRepository.java:66]: Yep yep, fixed

File: src/main/java/no/nav/pto/veilarbportefolje/opensearch/OpensearchQueryBuilder.java (hunk 5 above)
Comment (dragutin-nav) [reply src/main/java/no/nav/pto/veilarbportefolje/opensearch/OpensearchQueryBuilder.java:211]: Fixed

File: src/main/java/no/nav/pto/veilarbportefolje/ensligforsorger/EnsligeForsorgereRepository.java [hunk 7]
```diff
@@ -0,0 +1,296 @@
...
+            lagreOvergangsstonad(vedtakOvergangsstønadArbeidsoppfølging);
+        } else {
+            log.info("Vi støtter kun overgangstønad for enslige forsorgere. Fått: " + vedtakOvergangsstønadArbeidsoppfølging.getStønadstype());
```
[the remark this answers is not in the document]
Comment (dragutin-nav) [reply src/main/java/no/nav/pto/veilarbportefolje/ensligforsorger/EnsligeForsorgereRepository.java]: I agree, fixed

**Pull request merged**